src/browserLookups/localStorage.js

```diff
@@ -1,18 +1,27 @@
-let hasLocalStorageSupport;
-try {
-  hasLocalStorageSupport = window !== 'undefined' && window.localStorage !== null;
-  const testKey = 'i18next.translate.boo';
-  window.localStorage.setItem(testKey, 'foo');
-  window.localStorage.removeItem(testKey);
-} catch (e) {
-  hasLocalStorageSupport = false;
+let hasLocalStorageSupport = null;
+
+const localStorageAvailable = () => {
+  try {
+    hasLocalStorageSupport = window !== 'undefined' && window.localStorage !== null;
+    const testKey = 'i18next.translate.boo';
+    window.localStorage.setItem(testKey, 'foo');
+    window.localStorage.removeItem(testKey);
+  } catch (e) {
+    hasLocalStorageSupport = false;
+  }
+  return hasLocalStorageSupport;
+}
+
+const checkLocalStorage = () => {
+  if (hasLocalStorageSupport === null) hasLocalStorageSupport = localStorageAvailable();
```

Comment:
Why not moving this inside the upper function?

Reply:
Nice suggestion. Haven't thought about that.

Reply:
Will do.